features/192-asset-gear/implementation.md

```diff
@@ -0,0 +1,44 @@
+# Implementation Details and Decisions
+
+## Whether or not to use a gateway
+
+The evaluated gateway Minio does not support public access to object. All URLs are be signed even the underlying file is public read.
+
+The advantage offered by a gateway is allowing us to program against a specific interface like S3.
+However, the feature the Asset Gear needs is just presigning URL. It is possible to generate the presigned URL on our own.
+
+Therefore, the implementation is going to generate presigned request manually.
+
+## How to implement public and private asset
+
+S3 and GCE both supports private bucket with some public objects inside.
+Azure Storage, however, only supports bucket-level access control.
+The bucket is either private or public.
+The objects inherit the access control of the bucket and does not have any override.
```

Comment:
NFS as backend is a common requirement for Enterprise. See if you want to consider it too.

Reply:
We are not going to support NFS now. However, it is possible to support NFS via service such Minio in the future.